framework/src/main/java/org/checkerframework/framework/qual/HasQualifierParameter.java

```diff
@@ -0,0 +1,74 @@
+package org.checkerframework.framework.qual;
+
+import java.lang.annotation.Annotation;
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Target;
+
+/**
+ * This is a declaration annotation that applies to type declarations and packages. On a type, it
+ * means that the class conceptually takes a type qualifier parameter, though there is nowhere to
+ * write it because the class hard-codes a Java basetype rather than taking a type parameter.
+ * Writing {@code HasQualifierParameter} on a package is the same as writing it on each class in
+ * that package.
+ *
+ * <p>Writing {@code @HasQualifierParameter} on a type declaration has two effects.
+ *
+ * <ol>
+ *   <li>Invariant subtyping is used for occurrences of the type: no two occurrences of the type
+ *       with different qualifiers have a subtyping relationship.
+ *   <li>The polymorphic qualifier is the default for all occurrences of that type in a method
```

Comment:
Why only in method signatures? Why not in field types, local variables, etc.?

Reply:
That's what we think will lead to the least errors, but the default may need to be changed.  An individual checker could also choose a different defaulting scheme.

Reply:
generics.tex says `Within a class with a qualifier parameter, the default qualifier for uses of that class is the polymorphic qualifier.` without a limitation to just method signatures. Which one is it?